packages/smithy-client/src/split-header.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @param value - header string value.
+ * @returns value split by commas that aren't in quotes.
+ */
+export const splitHeader = (value: string): string[] => {
```

Comment:
This takes over `.split(",")` on the deserialization side. Splitting by commas is not enough, because commas within quotes do not count as delimiters, and once split, the items wrapped by double quotes should have those quotes removed.

Behavior specified here: https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restJson1/http-headers.smithy#L229-L232

```smithy
{
        id: "RestJsonInputAndOutputWithQuotedStringHeaders",
        documentation: "Tests responses with string list header bindings that require quoting",
        protocol: restJson1,
        code: 200,
        headers: {
            "X-StringList": "\"b,c\", \"\\\"def\\\"\", a"
        },
        params: {
            headerStringList: ["b,c", "\"def\"", "a"]
        }
}
```